### PR TITLE
Branch better find

### LINKED ISSUE #100: Implement better find for PROperty
# Overview
(Edited)
This issue aims to implement a better `find` command into PROperty. This better find command will allow users to search either names, phone numbers, emails, addresses or tags. Below is an example of hw I envision the find feature to work.

For example, let's say in PROperty, I have 5 individuals as follows:
1. John Doe, Phone No.: 123456789
2. John Doe, Phone No.: 12341234, Address: 24 Philip Street
3. Johnathan Andrew, Phone: 12341234
4. Jane Doe, Phone No.: 987654321
5. Andrew, Phone: 43214321, Address: 23 Philip Street

Find command expected outcomes:
- Case insensitivity: 
-- The find command should be case insensitive to searches.
--`find John Doe` or `find john doe` or find `JOHN DOE` should result in the following people listed out in the following order: 1. John Doe, 2. Jillian Doe, 3. Jane Doe.

- Specific searches:
-- When inputting a specific keyword, the individual should appear at the top of the list. However, the property agent may also like to see individuals with the same name/surname/address. Hence, order is important when showing the results.
-- `find s/John Doe s/24 Philip Street` will only return `John Doe` who lives on 24 Philip Street.

- General searches:
-- The property agent might want to search for individuals who live at the same general area, or with the same surname etc. Hence the find feature needs to accomodate a general search as well.
-- `find Philip Street` should result in the following people being displayed in the following order: 1. John Doe, 2. Andrew.
-- `find Doe` should result in the following people being displayed in the following order: 1. John Doe (Phone No.:123456789) , 2. John Doe (Phone No.: 12341234), 3. Jane Doe.

### PR DESCRIPTION
**Pull request by Vatinius**
Fixes #100

### CODE REVIEW

**Review by BuffWuff1712 — COMMENTED** [review]
Both general find and specific find works. Just need to include command instructions in help guide and include test cases.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -20,14 +21,18 @@ public class FindCommandParser implements Parser<FindCommand> {
      */
     public FindCommand parse(String args) throws ParseException {
         String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
+
+        if (trimmedArgs.startsWith(PREFIX_SPECIFIC_FIND.getPrefix()) && !trimmedArgs.isEmpty()) {
```
Comment (Vatinius) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:25]: Is this if statement acceptable, do you think?

Comment (BuffWuff1712) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java:25]: I think all's good

Comment (tayxuenye) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java:25]: Looks ok

File: src/test/java/seedu/address/model/person/ContainsGeneralKeywordsPredicateTest.java [hunk 2]
```diff
@@ -12,21 +12,24 @@
...
+public class ContainsGeneralKeywordsPredicateTest {
 
     @Test
     public void equals() {
         List<String> firstPredicateKeywordList = Collections.singletonList("first");
         List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
 
-        NameContainsKeywordsPredicate firstPredicate = new NameContainsKeywordsPredicate(firstPredicateKeywordList);
-        NameContainsKeywordsPredicate secondPredicate = new NameContainsKeywordsPredicate(secondPredicateKeywordList);
+        ContainsGeneralKeywordsPredicate firstPredicate =
```
Comment (tayxuenye) [inline src/test/java/seedu/address/model/person/ContainsGeneralKeywordsPredicateTest.java:22]: Good to see you renamed the NameContainsKeywordsPredicate class to suit current context

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 3]
```diff
@@ -14,28 +15,61 @@
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
+    public static final String SPECIFIC_FIND_PREFIX = "s/";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+            + "Example: " + COMMAND_WORD + " alice bob charlie\n"
```
Comment (tayxuenye) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:23]: Good that you extracted out COMMAND_WORD to make future code easier to edit

**Review by tayxuenye — APPROVED** [review]

**Comment by Vatinius** [comment]
Notable changes in user guide:
1. Further elaboration on `find s/` sub-feature.
2. Removed screenshot for find command as I found it out of place and inconsistent with other commands.

**Pull request merged**